### PR TITLE
Kit fixes and debug

### PR DESCRIPTION
**Pull request by SP4R0W**
@VargaDot review

### CODE REVIEW

File: blueprints/ranged/ranged.gd [hunk 1]
```diff
@@ -9,20 +9,32 @@ extends Node2D
...
 @export_enum("Automatic", "Semi", "Burst") var firing_mode:int
 
+@export var base_stats: RangedStatKit:
+	set(val):
+		if val:
+			mag_size = val.mag_size_modifier
+			max_ammo = val.max_ammo_modifier
+			reload_time = val.reload_time_modifier
+			reload_time_empty = val.reload_time_empty_modifier
+			fire_rate = val.fire_rate_modifier
```
Comment (VargaDot) [inline blueprints/ranged/ranged.gd:34]: smart

File: blueprints/ranged/ranged.gd [hunk 2]
```diff
@@ -124,27 +140,32 @@ func equip_stat_kit(kit:RangedStatKit) -> void:
...
-	mag_size *= int(1 - kit.mag_size_modifier / 100)
-	max_ammo *= int(1 - kit.max_ammo_modifier / 100)
-	reload_time *= (1 - kit.mag_size_modifier / 100)
-	reload_time_empty *= (1 - kit.mag_size_modifier / 100)
-	fire_rate *= (1 - kit.mag_size_modifier / 100)
+	mag_size = base_stats.mag_size_modifier
+	max_ammo = base_stats.max_ammo_modifier
+	reload_time = base_stats.reload_time_modifier
+	reload_time_empty = base_stats.reload_time_empty_modifier
+	fire_rate = base_stats.fire_rate_modifier
```
Comment (VargaDot) [inline blueprints/ranged/ranged.gd:158]: weapon stats should NOT revert back to base values, we have to keep in mind that a weapon will have more than one attachment.

this function should only remove the effects of the attachment.
```suggestion
	mag_size -= (mag_size - kit.mag_size_modifier)
	max_ammo -= (max_ammo - kit.max_ammo_modifier)
	reload_time -= (reload_time - kit.reload_time_modifier)
	reload_time_empty -= (reload_time_empty - kit.reload_time_empty_modifier)
	fire_rate -= (fire_rate - kit.fire_rate_modifier)
```

File: blueprints/kit/ranged_stat_kit.gd [hunk 3]
```diff
@@ -11,13 +11,13 @@ extends Resource
...
-@export var max_ammo_modifier:float
+@export var max_ammo_modifier:int
 ## The percentage by which to increase or decrease the reload speed.
 @export var reload_time_modifier:float
 ## The percentage by which to increase or decrease the reload speed when the magazine is empty.
 ## (reload time + reload time empty = actual reload time).
 @export var reload_time_empty_modifier:float
 ## The percentage by which to increase or decrease the rate of fire.
-@export var fire_rate_modifier:float
+@export var fire_rate_modifier:int
```
Comment (VargaDot) [inline blueprints/kit/ranged_stat_kit.gd:23]: documentation should be updated

**Review by VargaDot — CHANGES_REQUESTED** [review]
Overall good job, just needs a change of documentation and a fix

File: blueprints/kit/ranged_stat_kit.gd [hunk 4]
```diff
@@ -11,13 +11,13 @@ extends Resource
...
 ## The percentage by which to increase or decrease the maximum ammunition.
-@export var max_ammo_modifier:float
+@export var max_ammo_modifier:int
 ## The percentage by which to increase or decrease the reload speed.
 @export var reload_time_modifier:float
 ## The percentage by which to increase or decrease the reload speed when the magazine is empty.
 ## (reload time + reload time empty = actual reload time).
 @export var reload_time_empty_modifier:float
 ## The percentage by which to increase or decrease the rate of fire.
-@export var fire_rate_modifier:float
```
Comment (VargaDot) [inline blueprints/kit/ranged_stat_kit.gd:23]: why is fire_rate an int?

File: blueprints/ranged/ranged.gd (hunk 2 above)
Comment (VargaDot) [reply blueprints/ranged/ranged.gd:158]: check if the math is correct on my part too, if a kit decreases something, it has to be increased back and so on

Comment (SP4R0W) [reply blueprints/ranged/ranged.gd:158]: good point, will fix

File: blueprints/kit/ranged_stat_kit.gd (hunk 4 above)
Comment (SP4R0W) [reply blueprints/kit/ranged_stat_kit.gd:23]: my bad, will change

File: blueprints/kit/ranged_stat_kit.gd (hunk 3 above)
Comment (SP4R0W) [reply blueprints/kit/ranged_stat_kit.gd:23]: will update

File: blueprints/kit/ranged_stat_kit.gd [hunk 5]
```diff
@@ -10,14 +10,14 @@ extends Resource
...
+## The amount by which to increase or decrease the reload speed.
 @export var reload_time_modifier:float
-## The percentage by which to increase or decrease the reload speed when the magazine is empty.
+## The amount by which to increase or decrease the reload speed when the magazine is empty.
 ## (reload time + reload time empty = actual reload time).
 @export var reload_time_empty_modifier:float
-## The percentage by which to increase or decrease the rate of fire.
-@export var fire_rate_modifier:float
+## The amount by which to increase or decrease the rate of fire.
+@export var fire_rate_modifier: float
```
Comment (VargaDot) [inline blueprints/kit/ranged_stat_kit.gd:23]: i hate to be "that guy" but there's a space lmao

```suggestion
@export var fire_rate_modifier:float
```

**Review by VargaDot — CHANGES_REQUESTED** [review]

**Review by VargaDot — APPROVED** [review]

**Comment by VargaDot** [comment]
Good work @SP4R0W :3

**Pull request merged**